File: sandglass/sandlass.js
function sandlass( N, symbol ) {
    // 2n^2 - 1 <= N
    let n = Math.floor( Math.sqrt( ( N + 1 ) / 2 ) )
    let lineCount = 2 * n - 1
    let arr = new Array( 2 * n - 1 )
    for (var i = 0; i < lineCount; i++) {
        index = i >= n ? arr.length - i - 1 : i
        console.log( arr.fill( ' ' ).fill( symbol, index, arr.length - index ).join('') )
    }
    console.log( N - ( 2 * Math.pow( n, 2 ) - 1 ) )
}

sandlass( 19, '*' )

File: sandglass/sandlass.c
#include <stdio.h>
#include <math.h>

int main() {
    int N, n, lineCount, index;
    char symbol;
    scanf( "%d %c", &N, &symbol );
    n = sqrt( ( N + 1 ) / 2 );
    lineCount = 2 * n - 1;

    for( index = 0; index < lineCount; index++ ) {
        int offset = index >= n ? lineCount - index - 1 : index;
        for ( int jdenx = 0; jdenx < lineCount - offset; jdenx++ ) {
            if( jdenx < offset )
                printf( " " );
            else
                printf( "%c", symbol );
        }
        printf("\n");
    }
    /*
    @NOTE c lang is strong type
     */
    printf("%d\n",(int)( N - ( 2 * pow( (double)n, 2.0 ) - 1 ) ) );
    return 0;
}
